docs/mllib-clustering.md

```diff
@@ -718,6 +718,34 @@ sameModel = LDAModel.load(sc, "myModelPath")
 
 </div>
 
+## Bisecting k-means
+
+Bisecting k-means is a kind of [hierarchical clustering](https://en.wikipedia.org/wiki/Hierarchical_clustering).
+Hierarchical clustering is one of the most commonly used  method of cluster analysis which seeks to build a hierarchy of clusters.
+Strategies for hierarchical clustering generally fall into two types:
```

Comment:
Before this line (which is getting into details), it would be good to give a high-level description of the use of bisecting k-means.  E.g., "Bisecting K-means can often be much faster than regular K-means, but it will generally produce a different clustering."